packages/micropip/src/micropip/_micropip.py

```diff
@@ -483,17 +468,29 @@ def _list():
         >>> "regex" in package_list # doctest: +SKIP
         True
     """
-    packages = copy.deepcopy(INSTALLED_PACKAGES)
 
     # Add packages that are loaded through pyodide.loadPackage
+    packages = PackageDict()
+    for dist in importlib_distributions():
+        name = dist.name
+        version = dist.version
+        source = dist.read_text("PYODIDE_SOURCE")
+        if not source:
```

Comment:
```suggestion
        if source is None:
```
Maybe worth an inline comment that `source` is None if `PYODIDE_SOURCE` does not exis . It should never be empty.